frontend/src/components/contactsList.tsx

```diff
@@ -0,0 +1,90 @@
+import { useEffect, useState } from 'react'
+import { CreateAContactForm } from './contactsForm'
+import axios from 'axios'
+
+interface Contact {
+  id: string
+  firstName: string
+  lastName?: string
+  email?: string
+  phoneNumber?: string
+  notes?: string
+  source?: string
+  ownerID: string
+  relationships?: UserContactRelationship[]
+  importantDateEvent?: ImportantDateEvent[]
+}
+
+interface UserContactRelationship {
+  id: string
+  relationshipType: string
+}
+
+interface ImportantDateEvent {
+  id: string
+  date: string
+  eventType: string
+}
+
+export const ContactsList = () => {
+  const [showCreateAContactForm, setShowCreateAContactForm] = useState<boolean>(false)
+  const [contacts, setContacts] = useState<Contact[]>([])
+
+  const toggleContactForm = () => {
+    setShowCreateAContactForm(!showCreateAContactForm)
+  }
+
+  const handleContactCreate = (newContact: Contact) => {
+    console.log(newContact)
+    setContacts((prevState) => [...prevState, newContact])
+  }
+
+  useEffect(() => {
+    const getContacts = async () => {
+      try {
+        const response = await axios.get('http://localhost:8000/contacts')
+        const contactsData = response.data.contacts as Contact[]
+        setContacts(contactsData)
+      } catch (error) {
+        console.error(error)
```

Comment:
should we be doing something else besides logging the error to console?

Reply:
Probably display an error message to the user!

Reply:
are you going to add error states later? this simply logs an error to the console (which may not always be what we want to do)